developer-docs/kerberos.md

```diff
@@ -77,4 +77,140 @@ winrm:
+
+An Ubuntu container running OMI server and listening on both the HTTP and HTTPS WinRM endpoints is intended to simulate a Windows host in a non-Windows environment.
+
+On startup, the container is automatically domain joined to the Samba active directory and is reachable inside the UDN as `omiserver.bolt.test`. As with the Samba container, add an entry to `/etc/hosts` to be able to access it via DNS name from the Docker host environment.
+
+On startup, the Docker entrypoint script waits for the domain to be resolved via DNS and accessible before attempting to perform a domain join with `realm join` followed by `net ads join` (after configuring local Kerberos and Samba clients). The [`sssd`](https://docs.pagure.org/SSSD.sssd/) service is setup to use the [`ad provider`](https://docs.pagure.org/SSSD.sssd/users/ad_provider.html) so that it may look up domain accounts locally.
+
+To configure OMI server the `HTTP` service SPN is added to the `OMISERVER$` computer account in Active Directory, and the `sssd` service is restarted. The [OMI setup documentation](https://github.com/Microsoft/omi/blob/master/Unix/doc/setup-kerberos-omi.md#on-the-server-add-the-http-principal) covers this, but the actual scripts vary a bit since Samba is being used rather than Active Directory.
+
+The container performs a basic validation using `getent passwd administrator@BOLT.TEST` to verify the system is properly configured and domain joined. It then uses the `omicli` tool and the PowerShell cmdlet `Invoke-Command` to vet that the `bolt:bolt` account can authenticate properly using SPENGO. The domain Administrator account is then tested with the same tools to verify Kerberos authentication is working properly.
+
+###### External Ports
+
+* 45985 (tcp) - WinRM HTTP (internally 5985)
+* 45986 (tcp) - WinRM HTTPS (internally 5986)
+
+###### Interactive Shell Access
+
+To access the shell, use `/bin/bash` like:
+
+> docker-compose -f spec/docker-compose.yml exec omiserver /bin/bash
+
+Useful tooling on the instance includes:
+
+* [`host`](https://linux.die.net/man/1/host) - DNS lookup utility
+* [`klist`](https://web.mit.edu/kerberos/krb5-devel/doc/user/user_commands/klist.html) - check Kerberos tickets
+* [`realm`](https://www.systutorials.com/docs/linux/man/8-realm/) - manages enrollment in Kerberos realms and Active Directory domains
+* [`net`](https://www.samba.org/samba/docs/current/man-html/net.8.html) - designed to work like the `net` tool on Windows
+* [`pwsh`](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-linux?view=powershell-6) - PowerShell 6
+* [`omicli`](https://github.com/microsoft/omi/blob/master/Unix/cli/examples.txt) - client tool used to verify basic OMI server functionality
+
+#### TravisCI automated testing
+
+At present, Travis setup will:
+
+* Ensure that it can refer to itself as `samba-ad.bolt.test` and `omiserver.bolt.test` (the same name that the containesr refer to themselves inside the Docker UDN)
```

Comment:
```suggestion
* Ensure that it can refer to itself as `samba-ad.bolt.test` and `omiserver.bolt.test` (the same name that the containers refer to themselves inside the Docker UDN)
```